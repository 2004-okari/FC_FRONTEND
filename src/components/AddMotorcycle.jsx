import React from 'react'

const AddMotorcycle = () => {
  return (
    <div>AddMotorcycle</div>
  )
}

export default AddMotorcycle