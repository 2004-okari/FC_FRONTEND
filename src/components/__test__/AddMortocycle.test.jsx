import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import AddMotorcycle from '../AddMotorcycle';

const mockStore = configureStore([]);

describe('AddMotorcycle', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      bikes: {
        message: null,
        isLoading: false,
        error: null,
      },
    });
  });

  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <AddMotorcycle />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('form submission calls dispatch with correct data', () => {
    render(
      <Provider store={store}>
        <AddMotorcycle />
      </Provider>,
    );
  });
});
