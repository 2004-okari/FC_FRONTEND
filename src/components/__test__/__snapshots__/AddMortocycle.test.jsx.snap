// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMotorcycle renders correctly 1`] = `
<div
  className="container-div"
>
  <form
    className="create-container rounded-xl shadow-2xl border p-2"
    onSubmit={[Function]}
  >
    <h2
      className="h1"
    >
      Add New Car
    </h2>
    <label
      className="create-p-tag"
    >
      Car Name
    </label>
    <input
      autoComplete="name"
      className="create-input-tag"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      className="create-p-tag"
    >
      Car Image
    </label>
    <input
      className="create-input-tag"
      name="image"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      className="create-p-tag"
    >
      Description
    </label>
    <input
      className="create-input-tag"
      cols={4}
      name="description"
      onChange={[Function]}
      rows="10"
      value=""
    />
    <label
      className="create-p-tag"
    >
      Deposit
    </label>
    <input
      autoComplete="deposit"
      className="create-input-tag"
      min={0}
      name="deposit"
      onChange={[Function]}
      type="number"
      value=""
    />
    <label
      className="create-p-tag"
    >
      Finance Fees
    </label>
    <input
      autoComplete="finance_fee"
      className="create-input-tag"
      min={0}
      name="finance_fee"
      onChange={[Function]}
      type="number"
      value=""
    />
    <label
      className="create-p-tag"
    >
      Option to Purchase Fee
    </label>
    <input
      autoComplete="option_to_purchase_fee"
      className="create-input-tag"
      min={0}
      name="option_to_purchase_fee"
      onChange={[Function]}
      type="number"
      value=""
    />
    <label
      className="create-p-tag"
    >
      Total amount Payable
    </label>
    <input
      autoComplete="total_amount_payable"
      className="create-input-tag"
      min={0}
      name="total_amount_payable"
      onChange={[Function]}
      type="number"
      value=""
    />
    <label
      className="create-p-tag"
    >
      Duration
    </label>
    <input
      autoComplete="duration"
      className="create-input-tag"
      min={0}
      name="duration"
      onChange={[Function]}
      type="number"
      value=""
    />
    <button
      className="create-button-tag"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
